### PR TITLE
ENH: xz compression_args df.to_pickle(path, compression={'method':'xz','pr…

### LINKED ISSUE #52979: ENH: df.to_pickle xz lzma preset=level for compression compresslevel
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

I wish I could use pandas to save dataframe with maximum possible compression. xz is good, but no way now to set preset=9

### Feature Description

```
level = 9
df.to_pickle("name.pkl.xz", compression={'method': 'xz', 'compresslevel': level})
or
df.to_pickle("name.pkl.xz", compression={'method': 'xz', 'preset': level})
```

in /lib/python3.8/site-packages/pandas/io/common.py 828:
```
        elif compression == "xz":
            if ioargs.mode=='rb':
                handle = get_lzma_file()(handle, ioargs.mode)  # type: ignore[arg-type]
            else:
                handle = get_lzma_file()(handle, ioargs.mode, preset=level)
```

### Alternative Solutions

Other methods compression in pandas support compresslevel

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by segatrade**
…eset': level})
level = 9

- [x] closes #52979 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by twoertwein** [comment]
Thank you for the PR. You definitely will need the following:
- tests
- update the documentation https://github.com/pandas-dev/pandas/blob/8cf4ab491b06274f555d1ea305c172354208747a/pandas/core/shared_docs.py#L462 (and for decompression below that)
- and a whatsnew entry: doc/source/whatsnew/v2.1.0.rst

File: doc/source/whatsnew/v2.1.0.rst [hunk 1]
```diff
@@ -100,7 +100,7 @@ Other enhancements
 - Added ``engine_kwargs`` parameter to :meth:`DataFrame.to_excel` (:issue:`53220`)
 - Performance improvement in :func:`concat` with homogeneous ``np.float64`` or ``np.float32`` dtypes (:issue:`52685`)
 - Performance improvement in :meth:`DataFrame.filter` when ``items`` is given (:issue:`52941`)
--
+- Added ``**compression_args`` parameter to :meth:`DataFrame.to_pickle` (:issue:`52979`) for preset=level
```
Comment (twoertwein) [inline doc/source/whatsnew/v2.1.0.rst:103]: Maybe something along the lines of

> - Many read/to_* functions, such as :meth:`DataFrame.to_pickle` and :func:`read_csv`, support forwarding compression arguments to `lzma.LZMAFile` (:issue:`52979`)

**Comment by twoertwein** [comment]
> Thank you for advices and maintaining pandas! Seems done with checks and docs. Do I need add any new tests or it's ready?

Yes, you will need a test. Probably in https://github.com/pandas-dev/pandas/blob/main/pandas/tests/io/test_compression.py I'm not sure how to best test that the keywords are being forwarded: maybe one test with either to_pickle/read_csv and just pass a valid argument and then another test that passes an invalid keyword that raises an error in lzma to ensure the keyword is forwarded to it.

File: pandas/tests/io/test_compression.py [hunk 2]
```diff
@@ -252,6 +252,24 @@ def test_gzip_compression_level(obj, method):
...
         assert compressed_size_default < compressed_size_fast
 
+@pytest.mark.parametrize("method", ["to_pickle", "to_json", "to_csv"])
+def test_xz_compression_level_read(obj, method):
+    with tm.ensure_clean() as path:
+        getattr(obj, method)(path, compression="xz")
+        compressed_size_default = os.path.getsize(path)
+        getattr(obj, method)(path, compression={"method": "xz", "preset": 1})
+        compressed_size_fast = os.path.getsize(path)
+        assert compressed_size_default < compressed_size_fast
```
Comment (twoertwein) [inline pandas/tests/io/test_compression.py:273]: Great that you found a way to test the forwarded arguments directly! In that case, you can probably remove the `test_xz_compression_invalid_args`

**Comment by segatrade** [comment]
Thank you! Done. Seems now everything is ready for review.

**Review by twoertwein — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @segatrade

**Pull request merged**